openapi_codec/encode.py

```diff
@@ -23,15 +24,61 @@ def generate_swagger_object(document):
         swagger['schemes'] = [parsed_url.scheme]
 
     swagger['paths'] = _get_paths_object(document)
+    swagger['definitions'] = _get_definitions(document)
 
     return swagger
 
 
+def _get_definitions(document):
+
+    definitions = OrderedDict()
+    links = _get_links(document)
+
+    def get_field_def_data(field_item, defs):
+
+        definition_data = {
+            'type': 'object',
+            'properties': {}
+        }
+
+        if isinstance(field_item, coreschema.Object):
+            props = field_item.properties
+        else:
+            props = field_item.schema.properties
+
+        for f_name, f_schema in iter(props.items()):
+
+            if _get_field_type(f_schema) is 'object':
+                defs[f_name] = get_field_def_data(f_schema, defs)
+                definition_data['properties'][f_name] = {
+                    '$ref': '#/definitions/{}'.format(f_name)
+                }
+            else:
+                definition_data['properties'][f_name] = {
+                    'type': _get_field_type(f_schema),
```

Comment:
@tuffnatty no problems, that was looong hard day) I'm willing to fix it, so stay tuned. And yeah, thanks )